ethers-providers/src/transports/quorum.rs

```diff
@@ -190,15 +191,67 @@ impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
 
         let mut numbers = vec![];
         let mut errors = vec![];
+
+        // exit once required quorum weight is achieved
+        let required_weight = self.quorum_weight();
+        let mut weight_achieved = 0;
+
+        // timestamp of when we started
+        let start = SystemTime::now();
+
+        // wait until we've reached quorum
         while !queries.is_empty() {
             let (response, _index, remaining) = future::select_all(queries).await;
             queries = remaining;
             match response {
-                Ok(v) => numbers.push(v),
+                Ok(v) => {
+                    weight_achieved += v.0.weight;
+                    numbers.push(v);
+                    if weight_achieved >= required_weight {
+                        break;
+                    }
+                }
                 Err(e) => errors.push(e),
             }
         }
 
+        let quorum_reached_timestamp = SystemTime::now();
+
+        // Current grace period to wait for remaining requests is set to
+        // how long it took for us to reach quorum
+        let quorum_grace_period = quorum_reached_timestamp.duration_since(start)
+            .unwrap_or_default();
+        let timeout_timestamp = quorum_reached_timestamp.checked_add(quorum_grace_period)
+            .unwrap_or_else(|| SystemTime::now());
+        // try and wait for any remaining requests
+        while !queries.is_empty() {
+            let now = SystemTime::now();
+            // if timeout has been exceeded, break
+            if now >= timeout_timestamp {
+                break;
+            }
+            // how long to wait
+            let timeout_duration = match timeout_timestamp.duration_since(now) {
+                Ok(d) => d,
+                Err(_) => break,
+            };
+            match tokio::time::timeout(timeout_duration, future::select_all(queries)).await {
+                Ok((response, _index, remaining)) => {
+                    queries = remaining;
+                    match response {
+                        Ok(v) => {
+                            numbers.push(v);
+                        }
+                        Err(e) => errors.push(e),
+                    }
+
+                }
+                Err(_) => {
+                    break;
+                }
+            }
+        }
+
         numbers.sort_by(|(_, block_a), (_, block_b)| {
             // order by descending block number
```

Comment:
while we're at it, can you change the comments that mention block number because this is used for multiple rpc methods